File: src/io/logger.ts

export function log(msg: string) {
  console.log('[LOG]', msg);
}

export function logError(msg: string) {
  console.error('[ERR]', msg);
}

export function createEntry(level: string, msg: string) {
  return { level, msg, ts: Date.now() };
}

File: src/io/logger.c
#include "logger.h"
#include <stdio.h>

void log_msg(const char* msg) {
    printf("[LOG] %s\n", msg);
}

void log_error(const char* msg) {
    fprintf(stderr, "[ERR] %s\n", msg);
}

void log_init(void) {
    log_msg("logger initialized");
}
